### DISCUSSION TITLE
Running Nitro Server-side code at startup?

### DISCUSSION THREAD

**Discussion by robertpatrick** [nitrojs/nitro#1696] [Q&A]
We are building a new application in Nuxt 3.  As part of the application initialization, I need to execute code to that "discovers" all available types from a Kubernetes cluster.  Since this data is not client-specific and it is relatively expensive to put together, I would like to get the server to execute the code at startup (and make sure it is never runs in the browser) and then the browser would use fetch calls to get the data it needs.  The idea is that the server initiates running this code without waiting for a request from the application.

Is there a mechanism to achieve this goal?

**Comment by stafyniaksacha** [comment]
You can simply use a nitro plugin! https://nuxt.com/docs/guide/directory-structure/server#server-plugins

```ts
// server/plugins/my-plugin.ts
export default defineNitroPlugin((nitroApp) => {
  console.log('Nitro plugin', nitroApp)
})
```
